### ISSUE TITLE
Unnecessary rebuilding -- could be triggered by updating template timestamps?

### ISSUE THREAD

**Issue by dhbradshaw** [Kogia-sima/sailfish#46]
My project would recompile when I ran `cargo build` twice in a row.

When I removed sailfish, it started recognizing the fact that it was already compiled.

**Comment by Kogia-sima** [comment]
@dhbradshaw Welcome, and thank you for reporting the bug!

This seems to be cause by temporary artifact generated by sailfish. Could you paste the contents of your dep-info file? dep-info file is located at:

```
./target/<profile>/deps/<your crate name>-<hash>.d
```

**Comment by Kogia-sima** [comment]
I put a workaround on [`fix/incorrect-fingerprints/46`](https://github.com/Kogia-sima/sailfish/tree/fix/incorrect-fingerprints/46) branch. You can test this patch by replacing sailfish dependency with git repository.

```
[dependencies]
sailfish = { git = "https://github.com/Kogia-sima/sailfish", branch = "fix/incorrect-fingerprints/46" }
```

**Comment by ShadowItaly** [comment]
I have the exact same problem with the rebuilding. Your suggested fix is working for me! Thanks a lot great library by the way :)

**Comment by Kogia-sima** [comment]
@ShadowItaly Thanks for your feedback! Now I'll merge the branch into master soon.

**Comment by Kogia-sima** [comment]
This patch is included in version 0.3.1 release.
